src/geventhttpclient/connectionpool.py

```diff
@@ -204,8 +204,13 @@ def _create_tcp_socket(self, family, socktype, protocol):
                 family, socktype, protocol)
 
             if self.ssl_context_factory is None:
-                ssl_options = self.default_options.copy()
-                ssl_options.update(self.ssl_options)
+                ssl_options = {}
+                if not self.insecure:
+                    ssl_options = self.default_options.copy()
+                    ssl_options.update(self.ssl_options)
```

Comment:
As far as I understand it, "insecure" is for telling our client that we do not care if the server is secure (has a valid certificate, ...), but the server still might need from us a certificate file or other options. So not copying our secure default options there is fine but we still need the custom ones.